woocommerce/woocommerce-ios
# Fix incorrect order of the bar charts under specific conditions

Fix for https://github.com/woocommerce/woocommerce-ios/issues/887

⚠️  This is to be merged after https://github.com/woocommerce/woocommerce-ios/pull/870

## Before/After

<img src="https://user-images.githubusercontent.com/57791/57066652-0d5ffa00-6ccd-11e9-9ffb-a831c120c432.png" width="320" /> <img src="https://user-images.githubusercontent.com/57791/57066709-397b7b00-6ccd-11e9-89b8-58c51da54f5b.png" width="320" />

## Changes

The important commit is https://github.com/jeanregisser/woocommerce-ios/commit/6159dcee328576d431832554b63a81f1b3c47f3a

- Updated `handleOrderStatsItems` in `StatsStore` to remove any pre-existing items so the original order is preserved.
- Updated unit test to confirm the correct order.

## Testing

- Checkout the branch, run the tests
- Open the app
- Go to the custom tab
- Edit the range to be 1 year
- See the stats displayed in the correct order (left oldest, right newest)
- Edit the range to be 2 years (keeping the same end date)
- See the stats still displayed in the correct order (left oldest, right newest)

Update release notes:
